snapshot/restore/restore.go

```diff
@@ -199,14 +207,22 @@ func (c *copier) copyEntryInternal(ctx context.Context, e fs.Entry, targetPath s
 	}
 }
 
-func (c *copier) copyDirectory(ctx context.Context, d fs.Directory, targetPath string, onCompletion parallelwork.CallbackFunc) error {
+func (c *copier) copyDirectory(ctx context.Context, d fs.Directory, targetPath string, depth int, onCompletion parallelwork.CallbackFunc) error {
 	atomic.AddInt32(&c.stats.RestoredDirCount, 1)
 
+	if depth == -1 {
```

Comment:
this is confusing, can we introduce a helper that says:

func (c *copier) shouldRestoreAsDirEntry(depth, ...) {
  return depth >= options.RestoreDirEntryAtDepth
}

Reply:
I changed the logic. Hopefully the code is clearer. Maybe it doesn't need the helper.

Reply:
Done in e206514